kv_store/lww_kvs/kvs_memory_server.cpp

```diff
@@ -371,148 +383,290 @@ int main(int argc, char* argv[]) {
     } else if (pollitems[1].revents & ZMQ_POLLIN) {
       string new_server_ip = zmq_util::recv_string(&join_puller);
       cout << "Received a node join. New node is " << new_server_ip << ".\n";
-      master_node_t mnode = master_node_t(new_server_ip);
+      master_node_t new_node = master_node_t(new_server_ip, "M");
 
       // update hash ring
-      global_hash_ring.insert(mnode);
+      global_memory_hash_ring.insert(new_node);
       // instruct a random worker to send gossip to the new server! (2 phase)
-      size_t random_port = SERVER_PORT + rand()%MEMORY_THREAD_NUM + 1;
-      string worker_address = worker_node_t(ip, random_port).local_redistribute_addr_;
+      int tid = 1 + rand()%MEMORY_THREAD_NUM;
```

Comment:
Just curious: Why do we need a + 1?

Reply:
Because the worker thread number starts with 1 (not 0)